uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/components/leave-calendar.component.spec.js

```diff
@@ -0,0 +1,496 @@
+    'mocks/data/public-holiday-data',
+    'common/mocks/services/api/contact-mock',
+    'mocks/apis/absence-period-api-mock',
+    'mocks/apis/absence-type-api-mock',
+    'mocks/apis/public-holiday-api-mock',
+    'mocks/apis/option-group-api-mock',
+    'leave-absences/shared/config',
+    'leave-absences/my-leave/app'
+  ], function (angular, moment, _, helper, absencePeriodData, absenceTypeData, optionGroupMock, publicHolidayData) {
+    'use strict';
+
+    describe('leaveCalendar', function () {
+      var $componentController, $controller, $controllerProvider, $log, $q,
+        $rootScope, $timeout, controller, $provide, AbsencePeriod, OptionGroup,
+        PublicHoliday, sharedSettings;
+      var mockedCheckPermissions = mockCheckPermissionService();
+      var currentContact = {
+        id: CRM.vars.leaveAndAbsences.contactId,
+        role: 'staff'
+      };
+
+      beforeEach(module('common.mocks', 'leave-absences.templates', 'leave-absences.mocks', 'my-leave', function (_$provide_, _$controllerProvider_) {
+        $provide = _$provide_;
+        $controllerProvider = _$controllerProvider_;
+      }));
+
+      beforeEach(inject([
+        'AbsencePeriodAPIMock', 'AbsenceTypeAPIMock', 'PublicHolidayAPIMock', 'api.contact.mock',
+        function (AbsencePeriodAPIMock, AbsenceTypeAPIMock, PublicHolidayAPIMock, ContactAPIMock) {
+          $provide.value('AbsencePeriodAPI', AbsencePeriodAPIMock);
+          $provide.value('AbsenceTypeAPI', AbsenceTypeAPIMock);
+          $provide.value('PublicHolidayAPI', PublicHolidayAPIMock);
+          $provide.value('checkPermissions', mockedCheckPermissions);
+          $provide.value('api.contact', ContactAPIMock);
+        }
+      ]));
+
+      beforeEach(inject([
+        '$componentController', '$controller', '$log', '$q', '$rootScope', '$timeout',
+        'AbsencePeriod', 'OptionGroup', 'PublicHoliday', 'shared-settings', 'OptionGroupAPIMock',
+        function (_$componentController_, _$controller_, _$log_, _$q_, _$rootScope_,
+          _$timeout_, _AbsencePeriod_, _OptionGroup_, _PublicHoliday_, _sharedSettings_,
+          OptionGroupAPIMock) {
+          $componentController = _$componentController_;
+          $controller = _$controller_;
+          $log = _$log_;
+          $q = _$q_;
+          $rootScope = _$rootScope_;
+          $timeout = _$timeout_;
+          AbsencePeriod = _AbsencePeriod_;
+          PublicHoliday = _PublicHoliday_;
+          OptionGroup = _OptionGroup_;
+          sharedSettings = _sharedSettings_;
+
+          spyOn($log, 'debug');
+          spyOn($rootScope, '$emit').and.callThrough();
+          spyOn(AbsencePeriod, 'all');
+          spyOn(PublicHoliday, 'all').and.callThrough();
+          spyOn(OptionGroup, 'valuesOf').and.callFake(function (name) {
+            return OptionGroupAPIMock.valuesOf(name);
+          });
+
+          // Set 2016 as current period, because Calendar loads data only for the current period initially,
```

Comment:
I'm not sure i agree with this in the end. The multiline comment is usually used either to serve as some kind of header or to contain a *very long* and structured text. This is a simple comment that spans two line, that's all. It think it's better to use `//` as to mark the fact that it's a standard comment